src/Tokenizer/Token.php

```diff
@@ -77,15 +77,9 @@ public function __construct($token)
             $this->isArray = true;
             $this->id = $token[0];
             $this->content = $token[1];
-        } elseif (\is_string($token)) {
```

Comment:
why was this test removed?

Reply:
Bacause the (PHPDocs) type declaration (analyzed by PHPStan) guarantees that, in case `$token` is not an array, then it can only be a string, so the check is redundant.

Reply:
PHPDoc are not enforced during runtime like type arguments, people can still pass whatever and that is where this check is for, I opt to keep it until we typed down the argument using union in some new major

Reply:
So, Julien is for removing, SP is for keeping it, @keradus can you cast your deciding vote?

Reply:
I follow reasoning of @SpacePossum .
if type is not enforced on language level, we cannot assume the type, we should validate it

Reply:
Roger that.